index.js

```diff
@@ -0,0 +1,52 @@
+let button = document.getElementsByClassName("btn btn-danger")[0];
+let result = document.getElementById("result");
+let error = document.getElementById("error");
+
+function isNumber(value) {
+  return !isNaN(+value);
+}
+
+const formulas = {
+  addition: (a, b) => a + b,
+  subtraction: (a, b) => a - b,
+  multiplication: (a, b) => a * b,
+  division: (a, b) => a / b
+};
+
+const calculate = function() {
+  const inputs = [
+    "firstOperand",
+    "secondOperand",
+    "thirdOperand",
+    "fourthOperand"
+  ].map(id => +document.getElementById(id).value);
+  console.log("inputs", inputs);
+
+  if (inputs.filter(id => isNumber(id)).length !== inputs.length) {
+    error.textContent = "Please complete all the fields with numbers!";
+    return (result.textContent = "");
+  }
+
+  const operations = [];
+  document
+    .getElementsByName("operations")
+    .forEach(x => operations.push(x[x.selectedIndex].value));
+  console.log("operations", operations);
+
+  const output = inputs
```

Comment:
very nice that you used reduce, can you make the code more understandable by breaking it into small functions?

Reply:
if you are reffering to the part of the code from line 36 to 42: I tried to make it more understandable with a for loop
![2](https://user-images.githubusercontent.com/50665097/79050129-95b28300-7c30-11ea-93d3-eedc7a89a478.JPG)